source/material_model/rheology/strain_dependent.cc

```diff
@@ -415,7 +489,12 @@ namespace aspect
         if  (this->simulator_is_past_initialization() && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms))
           {
             const double edot_ii = std::max(sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),min_strain_rate);
-            const double e_ii = edot_ii*this->get_timestep();
+            double e_ii = edot_ii*this->get_timestep();
+
+            // Adjusting strain values to account for strain healing without exceeding an unreasonable range
+            if (healing_mechanism != no_healing)
+              e_ii = std::min(std::max(calculate_strain_healing(in,edot_ii,i),0.0),1.e5);
```

Comment:
This line may actually break a few tests. While I agree limiting the strain to have a minimum value of 0 is a good idea, very small negative strain can arise due to advection under/over shoots. We will see if any of the tests actually contain a negative value :-).

Reply:
Hi @erinheilman & @naliboff  John, Erin and I haven't met. I don't think that she is aware of the limiters that @sac-bsa (Mack) and I are working on for the various particle interpolation algorithms.  I don't know what you are working on in this PR, but it looks like you are trying to preserve the positivity of the strain, which is the sort of thing our limiters for particle interpolation are designed to do.  Our work is based on ideas that come from preventing overshoot and undershoot when using finite volume codes to model discontinuities in gas dynamics; i.e., shock waves.  Such methods go by many names, but one is positivity preserving; e.g., the density can _never_ be <= 0. Anyway Mack and I have implemented some limiters that will do this in other problems; e.g., the viscoelastic bending beam benchmark for the value of the variable 'beam', which is one of the particle properties.  John N. is working with us..

John, I am not clear what you meant by this comment

> While I agree limiting the strain to have a minimum value of 0 is a good idea, very small negative strain can arise due to advection under/over shoots.

Were you simply saying that this happens, or are you saying it is OK if this happens?  I agree with the former statement, but not with the latter.

Reply:
@egpuckett - The intent was to state that negative stress do arise, but not that it is ok or physical. The strain does not have an upper bound, but fantastic if I a limiter can be used with just a lower bound.